test/integration/cartridge_role_test.lua

```diff
@@ -289,3 +289,59 @@ g.test_set_export_from_require_role = function()
     local resp = server:http_request('get', '/metrics', {raise = false})
     t.assert_equals(resp.status, 200)
 end
+
+local function set_zones(zones)
+    local servers = {}
+    for k, v in pairs(zones) do
+        table.insert(servers, {uuid = k, zone = v})
+    end
+    return g.cluster.main_server.net_box:eval([[
+        local servers = ...
+        local ret, err = require('cartridge').admin_edit_topology({
+            servers = servers,
+        })
+        if ret == nil then
+            return nil, err
+        end
+        return true
+    ]], {servers})
+end
+
+g.test_zone_label = function()
+    local server = g.cluster.main_server
+    local cartridge_version = server.net_box:eval([[
+        local ver = require("cartridge").VERSION:gsub("%.","")
+        return tonumber(ver)
+    ]])
+    if cartridge_version >= 240 then
+        -- set server zone
+        local ok, err = set_zones({
+            [server.instance_uuid] = 'z1',
+        })
+        t.assert_equals({ok, err}, {true, nil})
+        assert_upload_metrics_config('/metrics')
+
+        local resp = server:http_request('get', '/metrics', {raise = false})
+        t.assert_equals(resp.status, 200)
+        for _, obs in pairs(resp.json) do
+            t.assert_equals(
+                'z1', obs.label_pairs['zone'],
+                ('Alias label is present in metric %s'):format(obs.metric_name)
+            )
+        end
+        -- test change server zone name in runtime
```

Comment:
Please split it to different tests

- `test_zone_label_present`
- `test_zone_label_changes_in_runtime`